src/WinRT.Runtime/ComWrappersSupport.net5.cs

```diff
@@ -446,7 +446,7 @@ static unsafe DefaultComWrappers()
             IUnknownVftblPtr = RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IUnknownVftbl), sizeof(IUnknownVftbl));
             (*(IUnknownVftbl*)IUnknownVftblPtr) = new IUnknownVftbl
             {
-                QueryInterface = (delegate* unmanaged[Stdcall]<IntPtr, ref Guid, out IntPtr, int>)qi,
+                QueryInterface = (delegate* unmanaged[Stdcall]<IntPtr, Guid*, IntPtr*, int>)qi,
```

Comment:
Mostly just curious, is ther a particular reason for using `IntPtr` in these function pointers? It would be "better" to eg. make this just be `<void*, Guid*, void**, int>`, etc. As in, the general recommendation especially from .NET 7 onwards is to stop using `IntPtr` as "a pointer", given it's technically a numeric type 🙂

Reply:
Can confirm that is the recommendation, `IntPtr` -> `void*`. See the following comment https://github.com/dotnet/runtime/pull/70297#discussion_r890495003.

Reply:
Thanks for calling this out.  If I were to change this, I would also want to change AddRef and Release for consistency, but those look to be used within the generated projections and thereby would break existing projections if I were to change it.  Due to that, I am deferring on this for now.